unittests/src/taintdag/tdag.cpp

```diff
@@ -0,0 +1,397 @@
+#include <catch2/catch.hpp>
+
+#include <algorithm>
+#include <cassert>
```

Comment:
Including `cassert` in testing code seems weird. What do we need this for?

Reply:
I think it is a left over from early attempts. I'll remove it.

Reply:
Working on this already.